### ISSUE TITLE
Epoxy attribute fields on a model cannot be changed once the model is added to a controller. Check that these fields are not updated, or that the assigned objects are not mutated, outside of the buildModels method. The only exception is if the change is made inside an Interceptor callback. Consider using an interceptor if you need to change a model after it is added to the controller and before it is set on the adapter. If the model is already set on the adapter then you must call `requestModelBuild` instead to recreate all models.

### ISSUE THREAD

**Issue by ElieBouNader** [airbnb/epoxy#1115]
i did everything and this issue still existing, any idea?

**Comment by ElieBouNader** [comment]
Please, does anyone faced this issue???

**Comment by hm-tamim** [comment]
You can't change of of any variable on bind method on controller class.

**Comment by mradzinski** [comment]
What's the issue to begin with? What are you trying to accomplish? 

The error message is quite self-descriptive, and it's telling you that you aren't supposed to mutate nor update models manually after they were added to the controller, but instead either use an interceptor or call `requestModelBuild` (which will diff and update the specific model that changed for you).

**Comment by adrielAd** [comment]
I also face this issue and I am using requestModelBuild() function after updating item at position 0.
If this is not correct way then can anyone here tell what is the best way to update any item and refreshing epoxyrecyclerview. 
I am using kotlin databinding

**Comment by Draketheb4dass** [comment]
I would love to see a use case of using an "interceptor", the docs do not say much about it.

**Comment by wiztensai** [comment]
i am also facing this issue. please, what have i do.

**Comment by kdgr** [comment]
i have a social feed screen in my project that i used epoxy for listing posts. users can like or unlike posts.  i'm facing this issue when the new posts are added to pagingcontroller.
 
is there any way?

**Comment by nmhung** [comment]
I was facing the same issue yesterday.
Now I managed to solve it.

TL;DR
The data for epoxy have to be different with the data source.

So, what I do is: 
1. Create source data and store in ViewModel.
2. Create a copy of source data every time epoxy requestModelBuild()
3. When I want to change the data of an item, I call the function in ViewModel to make change on the source data.
4. call epoxy recyclerview to requestModelBuild

**Comment by jimale** [comment]
Update your Model(Data class) and submit data, this will update the item that you want to update.
`override fun buildModels() {

        users.forEach {user ->
            UserModel_()
                .id(user.id)
                .name(user.name)
                .onClickListener { model, parentView, clickedView, position ->
                    user.name = "New updated name"
                }
                .addTo(this)
        }
    }`

**Comment by Unacademy-Nigam** [comment]
@jimale Can you share a sample?

**Comment by MoneyQianN** [comment]
In my case, I changed the priority that flag by @EpoxyAttribute when the bind holder run. so i try to new a local object copy from origin priority ,and render view use local object . wish this help you
